### PR TITLE
Adding large file write/read test for fileops

### PR DESCRIPTION
**Pull request by PoppySeedPlehzr**

### CODE REVIEW

**Comment by osqueryer** [comment]
:no_good: The commit 275f42c4f7bd466131a422a01e5a527ee544aee9 (Job results: [586](https://jenkins.osquery.io/job/osqueryPullRequestAudit/586/)) failed the code audit.

File: osquery/filesystem/tests/fileops_tests.cpp [hunk 1]
```diff
@@ -324,6 +326,34 @@ TEST_F(FileOpsTests, test_seekFile) {
...
+
+
+ {
+  std::vector<char> buffer(expected_len);
+  PlatformFile fd(path, PF_OPEN_EXISTING | PF_READ);
+  EXPECT_TRUE(fd.isValid());
+  auto read_len = fd.read(buffer.data(), expected_len);
+  printf("[+] Write %d bytes\n", (int)read_len);
+  EXPECT_EQ(expected_len, read_len);
+  EXPECT_EQ(0, ::memcmp(buffer.data(), expected.c_str(), expected_len));
```
Comment (theopolis) [inline osquery/filesystem/tests/fileops_tests.cpp]: Use a string's operator==

File: osquery/filesystem/tests/fileops_tests.cpp [hunk 2]
```diff
@@ -324,6 +326,34 @@ TEST_F(FileOpsTests, test_seekFile) {
...
+  EXPECT_EQ(expected_len,write_len);
+ }
+
+
+ {
+  std::vector<char> buffer(expected_len);
+  PlatformFile fd(path, PF_OPEN_EXISTING | PF_READ);
+  EXPECT_TRUE(fd.isValid());
+  auto read_len = fd.read(buffer.data(), expected_len);
+  printf("[+] Write %d bytes\n", (int)read_len);
```
Comment (theopolis) [inline osquery/filesystem/tests/fileops_tests.cpp]: Remove me

File: osquery/filesystem/tests/fileops_tests.cpp [hunk 3]
```diff
@@ -17,6 +17,8 @@
 #include "osquery/filesystem/fileops.h"
 #include "osquery/tests/test_util.h"
 
+#include <stdio.h>
```
Comment (theopolis) [inline osquery/filesystem/tests/fileops_tests.cpp]: Remove me

File: osquery/filesystem/tests/fileops_tests.cpp [hunk 4]
```diff
@@ -324,6 +324,30 @@ TEST_F(FileOpsTests, test_seekFile) {
...
+    EXPECT_EQ(expected_len, write_len);
+  }
+
+  {
+    std::vector<char> buffer(expected_len);
+    PlatformFile fd(path, PF_OPEN_EXISTING | PF_READ);
+    EXPECT_TRUE(fd.isValid());
+    auto read_len = fd.read(buffer.data(), expected_len);
+    EXPECT_EQ(expected_len, read_len);
+    EXPECT_TRUE(expected == std::string(buffer.data()));
```
Comment (theopolis) [inline osquery/filesystem/tests/fileops_tests.cpp]: Why not: `EXPECT_EQ(expected, std::string(buffer.data()));`?

**Pull request merged**